controller/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -0,0 +1,128 @@
+@RequestMapping("/certificate")
+public class GiftCertificateController {
+
+    private final GiftCertificateService certificateService;
+
+    private GiftCertificateController(GiftCertificateService certificateService) {
+        this.certificateService = certificateService;
+    }
+
+    /**
+     * Find certificate by id.
+     *
+     * @param id the id
+     * @return the response entity
+     * @throws ServiceException the service exception
+     */
+    @GetMapping("/{id}")
+    public ResponseEntity<CertificateDTO> find(@PathVariable(name = "id") Long id) throws ServiceException {
+        CertificateDTO certificateDTO = certificateService.find(id);
+        return ResponseEntity.ok(certificateDTO);
+    }
+
+    /**
+     * Add certificate.
+     *
+     * @param certificateDTO the certificate dto
+     * @return the response entity
+     * @throws ServiceException the service exception
+     */
+    @PostMapping(value = "/")
+    public ResponseEntity<CertificateDTO> add(@RequestBody CertificateDTO certificateDTO) throws ServiceException {
+        CertificateDTO result = certificateService.add(certificateDTO);
+        return ResponseEntity.ok(result);
+    }
+
+    /**
+     * Update certificate. Mark the fields that are not specified for updating null.
+     *
+     * @param certificateDTO the certificate dto
+     * @return the response entity
+     * @throws ServiceException the service exception
+     */
+    @PutMapping(value = "/", consumes = "application/json")
+    public ResponseEntity<CertificateDTO> update(@RequestBody CertificateDTO certificateDTO) throws ServiceException {
+        CertificateDTO result = certificateService.update(certificateDTO);
+        return ResponseEntity.ok(result);
+    }
+
+    /**
+     * Delete certificate by id.
+     *
+     * @param id the id
+     * @return the response entity
+     * @throws ServiceException the service exception
+     */
+    @DeleteMapping("/{id}")
+    public ResponseEntity<String> delete(@PathVariable(name = "id") Long id) throws ServiceException {
+        long result = certificateService.delete(id);
+        return result != -1L ? ResponseEntity.ok("Delete successful!") : ResponseEntity.ok("Delete unsuccessful!");
+    }
+
+    /**
+     * Find certificate by tag name.
+     *
+     * @param tagName the tag name
+     * @return the response entity
+     * @throws ServiceException the service exception
+     */
+    @GetMapping("/byName/{name}")
+    public ResponseEntity<List<CertificateDTO>> findByTag(@PathVariable(name = "name") String tagName)
+            throws ServiceException {
+        List<CertificateDTO> certificates = certificateService.findByTag(tagName);
+        return ResponseEntity.ok(certificates);
+    }
+
+    /**
+     * Search certificate by name or description part.
+     *
+     * @param part the part
+     * @return the response entity
+     * @throws ServiceException the service exception
+     */
+    @GetMapping("/byPart/{part}")
```

Comment:
Please read about REST naming convention, this endpoint is not correct

Reply:
fix